Calendar.cs

```diff
@@ -0,0 +1,77 @@
+﻿using iCal.PCL.DataModel;
+using iCal.PCL.Serialization;
+using RestSharp;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace GordonWare {
+
+    public enum GroupId {
+        Promotion,
+        Group1,
+        Group2,
+        Group3
+    }
+
+    public static class Calendar {
+
+        private static bool IsCurrentEvent(DateTime now, iCalVEvent ev) {
+            return ev.DTStart.Date == now.Date && ev.DTEnd.Date == now.Date &&
+                   ev.DTStart.TimeOfDay <= now.TimeOfDay && ev.DTEnd.TimeOfDay >= now.TimeOfDay;
+        }
+
+        private static bool ForGroup(iCalVEvent ev, GroupId groupId) {
+            return ev.Description.Contains($"INFO3 {Groups[(int) groupId]}");
+        }
+
+        private const string URL =
+            "http://edt-v2.univ-nantes.fr/calendar/ics?timetables[0]=20739&timetables[1]=20740&timetables[2]=20741&timetables[3]=20742";
```

Comment:
hardcoding the url doesn't look good imo, but reversing their "api" if we can call this like that would be tough

Reply:
I know, but it's meant to be public, so it is... The v2 doesn't require any login. So it's public to all... That's what they wanted